src/app.rs

```diff
@@ -324,26 +300,21 @@ impl<Ms, Mdl, INodes: IntoNodes<Ms> + 'static, GMs: 'static> App<Ms, Mdl, INodes
     #[allow(clippy::too_many_arguments)]
     pub(super) fn new(
         update: UpdateFn<Ms, Mdl, INodes, GMs>,
-        sink: Option<SinkFn<Ms, Mdl, INodes, GMs>>,
         view: ViewFn<Mdl, INodes>,
         mount_point: Element,
-        routes: Option<RoutesFn<Ms>>,
-        window_events: Option<WindowEventsFn<Ms, Mdl>>,
-        init_cfg: OptDynInitCfg<Ms, Mdl, INodes, GMs>,
         base_path: Rc<Vec<String>>,
     ) -> Self {
         let window = util::window();
         let document = window.document().expect("get window's document");
 
         Self {
-            init_cfg,
             cfg: Rc::new(AppCfg {
                 document,
                 mount_point,
                 update,
-                sink,
+                sink: None,
```

Comment:
Just note to not forget remove `sink`, `window_events` and `routes`.

Reply:
So I sorted that out. sink and window_events i don't recall making too much difference (my memory could be wrong there, so take with a grain of salt). Culling `routes`, though, cascaded into a few other code changes that come with broader consequences.